apps/space/app/compat/next/navigation.ts

```diff
@@ -0,0 +1,49 @@
+"use client";
+
+import { useMemo } from "react";
+import { useLocation, useNavigate, useParams as useParamsRR, useSearchParams as useSearchParamsRR } from "react-router";
+import { ensureTrailingSlash } from "./helper";
+
+export function useRouter() {
+  const navigate = useNavigate();
+  return useMemo(
+    () => ({
+      push: (to: string) => navigate(ensureTrailingSlash(to)),
+      replace: (to: string) => navigate(ensureTrailingSlash(to), { replace: true }),
+      back: () => navigate(-1),
+      forward: () => navigate(1),
+      refresh: () => {
+        location.reload();
+      },
+      prefetch: async (_to: string) => {
+        // no-op in this shim
+      },
+    }),
+    [navigate]
+  );
+}
+
+export function usePathname(): string {
+  const { pathname } = useLocation();
+  return pathname;
+}
+
+export function useSearchParams(): URLSearchParams {
+  const [searchParams] = useSearchParamsRR();
+  return searchParams;
+}
+
+export function useParams() {
+  return useParamsRR();
+}
+
+export function redirect(url: string): never {
+  if (typeof window !== "undefined") {
+    window.location.href = url;
+  }
+  throw new Error(`Redirecting to ${url}`);
+}
+
+export function notFound(): never {
+  throw new Error("Not found");
```

Comment:
@prateekshourya29 did you check that this works as expected? I would of expected you to throw a Response instead.